### DISCUSSION TITLE
How to call just beforeUnmount()?

### DISCUSSION THREAD

**Discussion by steven-twerdochlib** [vuejs/test-utils#2521] [Q&A] [answered]
Hey, 
I'm trying to unit test my code and keeping my tests specific to each method. How would I call beforeUnmount without calling any other methods and setting the appropriate props/data for the beforeUnmount() method?

Note: I tried the below but it seems the prop randomProp is still undefined in the call.

```
      await wrapper.setProps({ randomProp: ['propy1', 'propy2'] });
      await wrapper.vm.$options.beforeUnmount();
```

**Comment by cexbrayat** [comment, marked as answer]
Hi @steven-twerdochlib 

You can call `wrapper.unmount()` which will unmount the component and call the hooks.
